pyro/contrib/funsor/handlers/trace_messenger.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import funsor
+
+from pyro.poutine.subsample_messenger import _Subsample
+from pyro.poutine.trace_messenger import TraceMessenger as OrigTraceMessenger
+
+from pyro.contrib.funsor.handlers.primitives import to_funsor
+from pyro.contrib.funsor.handlers.runtime import _DIM_STACK
+
+
+class TraceMessenger(OrigTraceMessenger):
+    """
+    Setting ``pack_online=True`` packs online instead of after the fact,
+    converting all distributions and values to Funsors as soon as they are available.
+
+    Setting ``pack_online=False`` computes information necessary to do packing after execution.
+    Each sample site is annotated with a "dim_to_name" dictionary,
+    which can be passed directly to funsor.to_funsor.
+    """
+    def __init__(self, graph_type=None, param_only=None, pack_online=None):
```

Comment:
nit: Could you either set the default to `True` or add a comment in the docstring explaining the default value? Currently there is no way to tell except by reading source code.